streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -267,25 +277,32 @@ public long approximateNumEntries() {
 
     @Override
     public void flush() {
+        validateStoreOpen();
         lock.writeLock().lock();
         try {
+            validateStoreOpen();
             cache.flush(cacheName);
-            super.flush();
+            wrapped().flush();
         } finally {
             lock.writeLock().unlock();
         }
     }
 
     @Override
     public void close() {
+        lock.writeLock().lock();
         try {
-            flush();
-        } finally {
-            try {
-                super.close();
-            } finally {
-                cache.close(cacheName);
+            final LinkedList<RuntimeException> suppressed = executeAll(
+                () -> cache.flush(cacheName),
+                () -> cache.close(cacheName),
+                wrapped()::close
+            );
+            if (suppressed != null && !suppressed.isEmpty()) {
```

Comment:
Compare my comment below. We should avoid returning `null` to avoid the necessity of a `null` check.